### PR TITLE
Proposal: batch node access #5426

### LINKED ISSUE #5426: Add edge nodes in batches
**What would you like to be added/modified**:
I think it would be more elegant to accomplish this feature through documentation. Some modifications can also be made to the kubeedge code.

**Dynamic node addition**
- [ ]  If the node is newly created from the resource pool, we may consider using cloudinit to install edgecore.
- [ ]  Guide users to produce ISO images.

**Static node addition**

- [ ] Provide node adding scripts and usage instructions. 
- [ ] Ensure that the node to be added is in the same network area. Furthermore, the script must be performed on an execution computer that has SSH access to all of the nodes being added.
- [ ] Provide password and ssh-key login options.
- [ ] Consider using a remote ssh tool such as sshpass.

**Why is this needed**:

Currently, adding edge nodes requires users to log in to each node individually and manage them. Perhaps we can introduce node capabilities in batches.

### PR DESCRIPTION
**Pull request by WeiXingEsther**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #5426

**Special notes for your reviewer**:
This PR adds a new keadm batchjoin command to support batch access of edge nodes to the KubeEdge cluster. Key features include:
Batch Access Implementation: Automates the connection of multiple edge nodes using a YAML configuration file, which includes node IPs, CloudCore addresses, authentication tokens, and KubeEdge versions.
Automation Tool: Uses the github.com/melbahja/goph library for SSH connections and remote execution of keadm join.
Detailed Logging: This logs the status and results of each node registration.

**Does this PR introduce a user-facing change?**:

```release-note
- Introduced a new `keadm batchjoin` command for batch registration of edge nodes.
- Users can now use a YAML configuration file to automate the connection of multiple edge nodes to the KubeEdge cluster.
- Detailed logging of node registration status and results is now available.
```

### CODE REVIEW

File: docs/proposals/batch-node-access.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+- Create a YAML-formatted configuration file `config.yaml` to store all edge node information that needs to be accessed in batches.
+
+- **nodes**: Contains detailed information about each edge node.
+  - **ip**: The IP address of the edge node.
+  - **nodeName**: The name of each edge node, used to distinguish and identify each node in the cluster.
+  - **architecture**: The hardware architecture type of the edge node, such as `amd64`, `arm`, or `arm64`.
+  - **username**: The username required for SSH login to the edge node.
+  - **password**: The password required for SSH login to the edge node.
+
+- **keadm_params**: Contains detailed information for configuring edge node access.
```
Comment (WillardHu) [inline docs/proposals/batch-node-access.md:60]: We need the same flags as `keadm join` command

File: docs/proposals/batch-node-access.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+- Confirm the existence and configuration of the keadm tools on the control node, and if there is no tar.gz package that conforms to the naming convention, download the required keadm tools from the GitHub release.
+
+#### Configuration file preparation
+
+- Create a YAML-formatted configuration file `config.yaml` to store all edge node information that needs to be accessed in batches.
+
+- **nodes**: Contains detailed information about each edge node.
+  - **ip**: The IP address of the edge node.
+  - **nodeName**: The name of each edge node, used to distinguish and identify each node in the cluster.
```
Comment (WillardHu) [inline docs/proposals/batch-node-access.md:55]: The node name can be an optional parameter

File: docs/proposals/batch-node-access.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            log.Fatalf("Failed to batch join nodes: %v", err)
+        }
+    },
+}
+
+func init() {
+    rootCmd.AddCommand(batchjoinCmd)
+    batchjoinCmd.Flags().StringP("config", "c", "", "Path to the configuration file")
+    batchjoinCmd.MarkFlagRequired("config")
+}
```
Comment (WillardHu) [inline docs/proposals/batch-node-access.md:89]: This function definition is not very reasonable, you can refer to other commands

File: docs/proposals/batch-node-access.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+- Find the command definition file in the source code and add the new command processing logic.
+
+```go
+// batchjoinCmd represents the batchjoin command
+var batchjoinCmd = &cobra.Command{
+    Use:   "batchjoin",
+    Short: "Batch join multiple edge nodes to the KubeEdge cluster",
+    Long:  `batchjoin allows you to batch join multiple edge nodes to the KubeEdge cluster by providing a configuration file.`,
+    Run: func(cmd *cobra.Command, args []string) {
+        configFilePath, _ := cmd.Flags().GetString("config")
```
Comment (WillardHu) [inline docs/proposals/batch-node-access.md:78]: Not the correct way to get parameters, refer to other commands

Comment (WeiXingEsther) [reply docs/proposals/batch-node-access.md]: Thank you very much for your guidance and assistance. The four issues mentioned above have been adjusted, and I will be here to counter any adjustments you offer.

File: docs/proposals/batch-node-access.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+Batch access to edge nodes in the KubeEdge system:
+
+- Simplifies node access operations, improves deployment efficiency, and ensures system stability and reliability.
+
+
+
+## Proposal
+
+- **Configuration file support:** Supports configuration files in YAML format, containing information such as the CloudCore address, authentication token, KubeEdge version, etc. required for each node connection.
```
Comment (fujitatomoya) [inline docs/proposals/batch-node-access.md:19]: This is just a comment. this does not block any development for this PR. (IMO this feature is great enhancement.)

quick question, does it support the configuration that could be different from each edge node? in other words, can user describe the physical node specific configuration different from each other such as `containerRuntime`? i believe that this is out of scope for this feature, right? this feature is more like focusing on `common` configuration and join the bunch of cluster in the same network all together?

**Review by Shelley-BaoYue — COMMENTED** [review]
/lgtm

**Pull request merged**